File: leetcode/rookie/06_后移0.js
const moveZero = function (nums = []) {
  if (nums.length <= 1) {
    return nums;
  }
  let i = 0;
  for (let j = 0; j < nums.length; j++) {
    if (nums[j] !== 0) {
      let a = nums[j];
      nums[j] = nums[i];
      nums[i] = a;
      i++;
    }
  }
  return nums;
};

console.log(moveZero([0, 1, 0, 2, 3]));

File: leetcode/rookie/06_后移0.py
def moveZero(nums=[]):
    if len(nums) <= 1:
        return nums
    i, j = 0,0
    while j < len(nums):
        if nums[j] != 0:
            nums[i], nums[j] = nums[j], nums[i]
            i += 1
        j += 1
    return nums


nums = moveZero([1,2,0,2,0])

print(nums)
